### PR TITLE
Unified selection: Public API review

### PR DESCRIPTION
**Pull request by grigasp**
Closes https://github.com/iTwin/presentation/issues/692

### CODE REVIEW

File: packages/unified-selection/src/unified-selection/SelectionChangeEvent.ts [hunk 1]
```diff
@@ -42,71 +45,19 @@ export interface StorageSelectionChangeEventArgs {
...
-   */
-  public raiseEvent(args: StorageSelectionChangeEventArgs, storage: SelectionStorage) {
-    this._listeners.forEach((listener) => listener(args, storage));
-  }
-}
+export type StorageSelectionChangesListener = (
+  /** Arguments for the selection change event */
+  args: StorageSelectionChangeEventArgs,
+  /** Unused. Temporarily left to avoid a breaking change. */
+  _?: any,
```
[the remark this answers is not in the document]
Comment (grigasp) [reply packages/unified-selection/src/unified-selection/SelectionChangeEvent.ts:62]: Yes :) I'm not worried about someone using this - nobody is, at least at this moment. 

The problem that I'm solving here is TS build error when you try to pass `SelectionStorage` of this version to a component that expects a `SelectionStorage` from previous package version, e.g. `0.1`. We already have such situations in the wild - our test app, VCR test viewer, viewer are all passing this to `Presentation.initialize` - so we have to make sure we don't break their build.

File: packages/unified-selection/src/unified-selection/Selectable.ts [hunk 2]
```diff
@@ -15,7 +17,13 @@ export interface SelectableInstanceKey {
 }
 
 /**
- * A custom selectable
+ * A custom selectable, which has an identifier, knows how to loads its associated selectable instance keys
+ * and has custom data associated with it.
+ *
+ * An example of such selectable could be made from a grouping node, which groups some instances together. The selectable's
+ * identifier could be some GUID associated with the node, and it would know how to load grouped instance keys from the grouping node.
```
[the remark this answers is not in the document]
Comment (grigasp) [reply packages/unified-selection/src/unified-selection/Selectable.ts:24]: I agree the first part is better, but not sure about the end - `loadInstanceKeys` is not given a guid. What do you think about the following?

> An example of such selectable could be an instance grouping node:
> - `identifier` could be a GUID, associated with the node,
> - `loadInstanceKeys` would know how to load grouped instance keys from the node,
> - `data` could be set to the node itself.

[the remark this answers is not in the document]
Comment (Yato333) [reply packages/unified-selection/src/unified-selection/Selectable.ts:24]: Yes that works as well. Basically I only didn't like the "some GUID" part of this 😄

[the remark this answers is not in the document]
Comment (grigasp) [reply packages/unified-selection/src/unified-selection/Selectable.ts]: updated

**Review by Yato333 — APPROVED** [review]

**Pull request merged**